### ISSUE TITLE
Bogus error 'Method is provate'

### ISSUE THREAD

**Issue by svenschmidt75** [intellij-rust/intellij-rust#3436]
## Environment

* **Intellij-Rust plugin version:** 0.2.92.2116-191
* **Rust toolchain version:** rustc 1.32.0 (9fda7c223 2019-01-16)
* **IDE name and version:** CLion 2019.EAP1
* **Operating system:** Ubuntu 18.04

## Problem description

Get https://github.com/svenschmidt75/Rust/tree/master/NeuralNetwork, commit 8194bd9307840b6186f60e69845d3dae73ab084a.
Open file fc_layer.rs and see red squigglies for methods being private. Those methods are
trait methods implemented in the derived type.
The code compiles just fine.
![screenshot from 2019-02-23 07-34-23](https://user-images.githubusercontent.com/17226522/53287725-8414d000-373d-11e9-9f33-e7adddc17274.png)

**Comment by Kobzol** [comment]
This is a smaller reproduction:
```rust
use foo:: {Foo, Trait};

mod foo {
    pub struct Foo;
    impl Foo {
        fn get(&self) { println!("struct"); }
    }
    
    pub trait Trait {
        fn get(&self);
    }
    impl Trait for Foo {
        fn get(&self) { println!("trait"); }
    }
}

fn main() {
    let f = foo::Foo;
    f.get();
}
```
`f.get()` gets resolved to `Foo::get` instead of `Trait::get`. Even though inherent `impl`s have large priority than trait `impl`s, this is not true if the inherent method is not accessible (here because of visibility). If the inherent method was public, it would be used, but because it is private, the method from the trait is used.

This assumption is therefore not always valid: https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt#L1218. @vlad20012 can we use visibility information in name resolution here?
